users/postgres/groups.go

```diff
@@ -154,15 +154,23 @@ func (gr groupRepository) RetrieveAllWithAncestors(ctx context.Context, groupID
 		return users.GroupPage{}, errors.Wrap(errRetrieveDB, err)
 	}
 
-	q := fmt.Sprintf(`WITH RECURSIVE subordinates AS (
-						SELECT id, owner_id, parent_id, name, description, metadata
-						FROM groups
-						WHERE id = :id 
-						UNION
-							SELECT groups.id, groups.owner_id, groups.parent_id, groups.name, groups.description, groups.metadata
-							FROM groups 
-							INNER JOIN subordinates s ON s.id = groups.parent_id %s
-					) SELECT * FROM subordinates ORDER BY id LIMIT :limit OFFSET :offset`, mq)
+	subQuery := fmt.Sprintf("SELECT id, owner_id, parent_id, name, description, metadata from groups WHERE 1=1 %s", mq)
+	q := fmt.Sprintf("%s ORDER BY id LIMIT :limit OFFSET :offset", subQuery)
+	cq := fmt.Sprintf("SELECT COUNT(*) FROM groups WHERE 1=1 %s", mq)
```

Comment:
I understand the "hack", but I'm not sure that I like it. This can be solved with a simple `if` which is easier to understand, right?